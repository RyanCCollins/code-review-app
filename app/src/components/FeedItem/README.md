## FeedItem Component
A component that ...

### Example

```js
<FeedItem />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
