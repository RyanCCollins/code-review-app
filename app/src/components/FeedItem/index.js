import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Heading from 'grommet/components/Heading';
import Paragraph from 'grommet/components/Paragraph';
import Footer from 'grommet/components/Footer';
import ChatIcon from 'grommet/components/icons/base/Chat';
import { Link } from 'react-router';

const FeedItem = ({
  project,
}) => (
  <div className={styles.feedItem}>
    <Heading>
      {project.title}
    </Heading>
    <Paragraph>
      {project.description}
    </Paragraph>
    <Footer>
      <span className={styles.commentLinks}>
        <Link to={`/projects/${project.slug}`}>
          <ChatIcon size="small" />
          {project.comments.length}
        </Link>
      </span>
    </Footer>
  </div>
);

FeedItem.propTypes = {
  project: PropTypes.object.isRequired,
};

export default cssModules(FeedItem, styles);
