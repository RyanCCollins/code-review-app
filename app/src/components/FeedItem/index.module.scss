.feedItem {
  padding: 12px 16px;
  background: #FFF;
  margin-bottom: 14px;
  border-radius: 1px;
  border: 1px solid #E0E0E0;
  box-shadow: 0 1px 2px #E0E0E0;
  max-width: 610px;
}

.commentLinks {
  font-size: 16px;
  padding: 7px 8px;
  color: #9E9E9E;
}
