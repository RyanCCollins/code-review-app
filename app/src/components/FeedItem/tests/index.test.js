import FeedItem from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<FeedItem />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <FeedItem
        project={{
          title: 'Hello World',
          description: 'A project',
          slug: 'hello-world',
          comments: [
            {
              id: 0,
              content: 'Hello World',
            },
          ],
          user: {
            name: 'Ryan Collins',
          },
        }}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
