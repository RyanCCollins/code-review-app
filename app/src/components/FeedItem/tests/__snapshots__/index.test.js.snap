exports[`<FeedItem /> should render with default props 1`] = `
<undefined>
  <function Heading() {
      (0, _classCallCheck3.default)(this, Heading);
      return (0, _possibleConstructorReturn3.default)(this, (Heading.__proto__ || (0, _getPrototypeOf2.default)(Heading)).apply(this, arguments));
    }
    tag="h3">
    Hello World
  </function Heading() {
      (0, _classCallCheck3.default)(this, Heading);
      return (0, _possibleConstructorReturn3.default)(this, (Heading.__proto__ || (0, _getPrototypeOf2.default)(Heading)).apply(this, arguments));
    }>
  <function Paragraph() {
      (0, _classCallCheck3.default)(this, Paragraph);
      return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
    }>
    A project
  </function Paragraph() {
      (0, _classCallCheck3.default)(this, Paragraph);
      return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
    }>
  <function Footer(props, context) {
      (0, _classCallCheck3.default)(this, Footer);
  
      var _this = (0, _possibleConstructorReturn3.default)(this, (Footer.__proto__ || (0, _getPrototypeOf2.default)(Footer)).call(this, props, context));
  
      _this._alignMirror = _this._alignMirror.bind(_this);
      _this._onResize = _this._onResize.bind(_this);
      return _this;
    }
    direction="row"
    responsive={false}>
    <span
      className={undefined}>
      <function (props, context, updater) {
            // This constructor gets overridden by mocks. The argument is used
            // by mocks to assert on what gets mounted.
      
            if (process.env.NODE_ENV !== 'production') {
              process.env.NODE_ENV !== 'production' ? warning(this instanceof Constructor, 'Something is calling a React component directly. Use a factory or ' + 'JSX instead. See: https://fb.me/react-legacyfactory') : void 0;
            }
      
            // Wire up auto-binding
            if (this.__reactAutoBindPairs.length) {
              bindAutoBindMethods(this);
            }
      
            this.props = props;
            this.context = context;
            this.refs = emptyObject;
            this.updater = updater || ReactNoopUpdateQueue;
      
            this.state = null;
      
            // ReactClasses doesn't have constructors. Instead, they use the
            // getInitialState and componentWillMount methods for initialization.
      
            var initialState = this.getInitialState ? this.getInitialState() : null;
            if (process.env.NODE_ENV !== 'production') {
              // We allow auto-mocks to proceed as if they're returning null.
              if (initialState === undefined && this.getInitialState._isMockFunction) {
                // This is probably bad practice. Consider warning here and
                // deprecating this convenience.
                initialState = null;
              }
            }
            !(typeof initialState === 'object' && !Array.isArray(initialState)) ? process.env.NODE_ENV !== 'production' ? invariant(false, '%s.getInitialState(): must return an object or null', Constructor.displayName || 'ReactCompositeComponent') : _prodInvariant('82', Constructor.displayName || 'ReactCompositeComponent') : void 0;
      
            this.state = initialState;
          }
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/projects/hello-world">
        <function Icon() {
            (0, _classCallCheck3.default)(this, Icon);
            return (0, _possibleConstructorReturn3.default)(this, (Icon.__proto__ || (0, _getPrototypeOf2.default)(Icon)).apply(this, arguments));
          }
          a11yTitleId="chat-title"
          responsive={true}
          size="small" />
        1
      </function (props, context, updater) {
            // This constructor gets overridden by mocks. The argument is used
            // by mocks to assert on what gets mounted.
      
            if (process.env.NODE_ENV !== 'production') {
              process.env.NODE_ENV !== 'production' ? warning(this instanceof Constructor, 'Something is calling a React component directly. Use a factory or ' + 'JSX instead. See: https://fb.me/react-legacyfactory') : void 0;
            }
      
            // Wire up auto-binding
            if (this.__reactAutoBindPairs.length) {
              bindAutoBindMethods(this);
            }
      
            this.props = props;
            this.context = context;
            this.refs = emptyObject;
            this.updater = updater || ReactNoopUpdateQueue;
      
            this.state = null;
      
            // ReactClasses doesn't have constructors. Instead, they use the
            // getInitialState and componentWillMount methods for initialization.
      
            var initialState = this.getInitialState ? this.getInitialState() : null;
            if (process.env.NODE_ENV !== 'production') {
              // We allow auto-mocks to proceed as if they're returning null.
              if (initialState === undefined && this.getInitialState._isMockFunction) {
                // This is probably bad practice. Consider warning here and
                // deprecating this convenience.
                initialState = null;
              }
            }
            !(typeof initialState === 'object' && !Array.isArray(initialState)) ? process.env.NODE_ENV !== 'production' ? invariant(false, '%s.getInitialState(): must return an object or null', Constructor.displayName || 'ReactCompositeComponent') : _prodInvariant('82', Constructor.displayName || 'ReactCompositeComponent') : void 0;
      
            this.state = initialState;
          }>
    </span>
    <undefined
      user={
        Object {
          "name": "Ryan Collins"
        }
      } />
  </function Footer(props, context) {
      (0, _classCallCheck3.default)(this, Footer);
  
      var _this = (0, _possibleConstructorReturn3.default)(this, (Footer.__proto__ || (0, _getPrototypeOf2.default)(Footer)).call(this, props, context));
  
      _this._alignMirror = _this._alignMirror.bind(_this);
      _this._onResize = _this._onResize.bind(_this);
      return _this;
    }>
</undefined>
`;
